Tests/Test1/ThreadSafePriorityQueue/ThreadSafePriorityQueue.cs

```diff
@@ -0,0 +1,64 @@
+﻿namespace ThreadSafePriorityQueue;
+
+/// <summary>Thread-safe priority queue.</summary>
+/// <typeparam name="TV">The type of elements.</typeparam>
+/// <typeparam name="TP">The type of priority. Inheritance from <see cref="IComparable"/> required.</typeparam>
+public class ThreadSafePriorityQueue<TV, TP>
+    where TP : IComparable
+    where TV : struct
+{
+    private readonly SortedDictionary<TP, Queue<TV>> _queueMap = new();
+    private int _elementsCount;
+
+    /// <summary>Returns elements count in queue.</summary>
+    public int Size
+    {
+        get
+        {
+            lock (_queueMap)
+            {
+                return _elementsCount;
+            }
+        }
+    }
```

Comment:
```suggestion
    public int Size => _elementsCount;
```
Это лучше было свойством сделать, метод предполагает содержательные вычисления.

И здесь `lock` не нужен. `int` атомарен (а значит другие потоки нам не помешают прочитать корректное значение), и мы не меняем значение переменной (а значит наш поток не помешает другим).
Возможно, мы прочтем неактуальное значение, но этого достаточно по заданию.